viadot/sources/databricks.py

```diff
@@ -0,0 +1,396 @@
+        import pandas as pd
+        databricks = Databricks()
+        list = [{"id":"1", "name":"Joe"}]
+        df = pd.DataFrame(list)
+        spark_df = databricks._pandas_df_to_spark_df(df)
+        ```
+        Returns:
+            spark.DataFrame: The resulting Spark DataFrame.
+        """
+        spark_df = self.session.createDataFrame(df)
+        return spark_df
+
+    def _spark_df_to_pandas_df(self, spark_df: spark.DataFrame) -> pd.DataFrame:
+        """
+        Convert a Spark DataFrame to a Pandas DataFrame.
+
+        Args:
+            df (spark.DataFrame): The Spark DataFrame to be converted to a Pandas DataFrame.
+
+        Example:
+        ```python
+        from viadot.sources import Databricks
+        databricks = Databricks()
+        list = [{"id":"1", "name":"Joe"}]
+        df = pd.DataFrame(list)
+        spark_df = databricks._pandas_df_to_spark_df(df)
+        pandas_df = databricks._spark_df_to_pandas_df(spark_df)
+        ```
+        Returns:
+            pd.DataFrame: The resulting Pandas DataFrame.
+        """
+        return spark_df.toPandas()
+
+    def run(
+        self, query: str, fetch_type: Literal["record", "dataframe"] = "record"
+    ) -> Union[spark.DataFrame, pd.DataFrame, bool]:
+        """
+        Execute an SQL query.
+
+        Args:
+            query (str): The query to execute.
+            fetch_type (Literal[, optional): How to return the data: either
+            in the default record format or as a Pandas DataFrame. Defaults to "record".
+
+        Example:
+        ```python
+        from viadot.sources import Databricks
+        databricks = Databricks()
+        query_result = databricks.run("SELECT * FROM schema.table_1")
+        ```
+        Returns:
+            Union[spark.DataFrame, pd.DataFrame, bool]: Either the result set of a query or,
+            in case of DDL/DML queries, a boolean describing whether
+            the query was excuted successfuly.
+        """
+        allowed_fetch_type_values = ["record", "dataframe"]
+        if fetch_type not in allowed_fetch_type_values:
+            raise ValueError(
+                f"Only the values {allowed_fetch_type_values} are allowed for 'fetch_type'"
+            )
+
+        query_clean = query.upper().strip()
+        query_keywords = ["SELECT", "SHOW", "PRAGMA", "DESCRIBE"]
+
+        query_result = self.session.sql(query)
+
+        if any(query_clean.startswith(word) for word in query_keywords):
+            if fetch_type == "record":
+                result = query_result
+            else:
+                result = self._spark_df_to_pandas_df(query_result)
+        else:
+            result = True
+
+        return result
+
+    def _check_if_table_exists(self, table_name: str) -> bool:
+        split_name = table_name.split(".")
+        does_exist = self.session._jsparkSession.catalog().tableExists(
+            split_name[0], split_name[1]
+        )
+        return does_exist
```

Comment:
Refer back to the comment about splitting table and schema as separate parameters.